### PR TITLE
Add support for prow tracing

### PR DESCRIPTION
**Pull request by howardjohn**
This enables our root entrypoint to emit trace spans. Currently, this will do nothing since no repo has the tracing script.

The intent to source from `./common/` is that the script will need to exist in the repo anyways so that it can be used for scripts in repos, so we can just use that rather than embedding it.

Once the common-files script is added, this will still do nothing until prow jobs are enrolled with the various env vars required to enable tracing. Without that, the tracing calls are NOPs.

### CODE REVIEW

File: docker/build-tools/prow-entrypoint.sh [hunk 1]
```diff
@@ -66,64 +77,67 @@ log "Starting test..."
...
+  }
+
+  trap cleanup EXIT
+
+  # Always try to authenticate to GCR and AR.
+  gcloud auth configure-docker "${GCP_REGISTRIES}" -q || true
 }
 
-trap cleanup EXIT
+tracing::run "docker" run_docker
```
Comment (jwendell) [inline docker/build-tools/prow-entrypoint.sh:125]: this call to `run_docker` (and similar calls below) simply do not run as `tracing::run` does nothing, right?

Comment (howardjohn) [reply docker/build-tools/prow-entrypoint.sh:125]: :facepalm: thanks!!

Comment (howardjohn) [reply docker/build-tools/prow-entrypoint.sh:127]: Fixed

**Comment by ericvn** [comment]
LGTM

**Review by jwendell — APPROVED** [review]

**Pull request merged**